lib/src/main/scala/spinal/lib/bus/amba3/ahblite/sim/AhbAttributes.scala

```diff
@@ -0,0 +1,45 @@
+package spinal.lib.bus.amba3.ahblite.sim
+
+object AhbAttributes {
+  def apply(attr: Hburst): AhbAttributes = new AhbAttributes()(attr)
+  def apply(attr: Hmastlock): AhbAttributes = new AhbAttributes()(attr)
+  def apply(attr: Hprot): AhbAttributes = new AhbAttributes()(attr)
+  def apply(attr: Hsize): AhbAttributes = new AhbAttributes()(attr)
+  def apply(attr: Htrans): AhbAttributes = new AhbAttributes()(attr)
+}
+
+/** Set of AHB attributes.
+  *
+  * Example:
+  *
+  * {{{
+  * AhbAttributes(Hmastlock.HIGH)(Hprot.opcodeFetch.withNonCacheable)
+  * }}}
+  */
+class AhbAttributes(
+    val hburst: Option[Int] = None,
+    val hmastlock: Option[Boolean] = None,
+    val hprot: Option[Int] = None,
+    val hsize: Option[Int] = None,
+    val htrans: Option[Int] = None
+) {
+  def copy(
+      hburst: Option[Int] = hburst,
+      hmastlock: Option[Boolean] = hmastlock,
+      hprot: Option[Int] = hprot,
+      hsize: Option[Int] = hsize,
+      htrans: Option[Int] = htrans
+  ): AhbAttributes = new AhbAttributes(
+    hburst,
+    hmastlock,
+    hprot,
+    hsize,
+    htrans
+  )
+
+  def apply(attr: Hburst): AhbAttributes = copy(hburst = Some(attr))
```

Comment:
Funny pattern XD

Reply:
Isn't it too unidiomatic?

Reply:
So, it is a bit too much i think ^^
As instead you can use case class and named argument : 

```scala
case class AhbAttributes(
    val hburst: Option[Int] = None,
    val hmastlock: Option[Boolean] = None,
    val hprot: Option[Int] = None,
    val hsize: Option[Int] = None,
    val htrans: Option[Int] = None
) 

val x = AhbAttributes(hprot=Some(2))
val y = x.copy(hsize=Some(32), htrans=Some(64))
```

Reply:
Indeed I started with a case class, then I wanted to add `def hburst(value)` and it obviously did not work so I removed `case`, then I realized I wanted to access values by these names so I left things half done etc.

Now I've added `withAttribute` and put `case class`

Reply:
Indeed, as there is implicit conversion from `Hanything` to `Int`, the `withAttr` would work anyway. Maybe I should also remove attribute classes which do not enable to build values… Yes, I'll definitely clean things here ^^'

Reply:
Actually I am still wondering (since I wrote the first commit) if gathering attributes in a struct is really useful? I mean, I the PR I show the version with it and without it, and sometimes parts of it are ignored, maybe I can just remove this class after all… And this notion is not in the protocol…

Reply:
> Actually I am still wondering (since I wrote the first commit) if gathering attributes in a struct is really useful? I mean, I the PR I show the version with it and without it, and sometimes parts of it are ignored, maybe I can just remove this class after all… And this notion is not in the protocol…

Ahh that's very much usage dependent, as you feel it. i don't realy have an opinon on it. Just that sometime, keeping things more raw, isn't a bad thing ^^